main.nf

```diff
@@ -0,0 +1,54 @@
+// Ensure DSL2
+nextflow.enable.dsl = 2
+
+//url for exmaple CWL workflow
+params.cwl_url = "https://github.com/CRI-iAtlas/iatlas-workflows/blob/develop/Cibersort/workflow/iatlas_workflow/api_workflow.cwl"
+//example params file
+params.input_file = "syn51089849"
+
+//get input yaml file from synapse
+//downloads synapse file given Synapse ID
+process SYNAPSE_GET {
+
+    container "sagebionetworks/synapsepythonclient:v2.7.0"
+    
+    secret 'SYNAPSE_AUTH_TOKEN'
+
+    input:
+    val(syn_id)
+
+    output:
+    path('*.yaml')
+
+    script:
+    """    
+    synapse get ${syn_id}
+    shopt -s nullglob
+    for f in *\\ *; do mv "\${f}" "\${f// /_}"; done
+    """
+}
+
+//runs cwl workflow using url and params provided
+process RUN_CWL_WORKFLOW {
```

Comment:
I just realized that we'll need to enable Docker-in-Docker for this approach to work (assuming that the iAtlas workflows leverage Docker).

Reply:
I haven't seen any workflows that use docker yet, I will confirm.

Reply:
The workflows do use docker, but the current implementation does work for docker-in-docker